include/boost/geometry/algorithms/detail/distance/geometry_to_segment_or_box.hpp

```diff
@@ -316,6 +345,48 @@ class geometry_to_segment_or_box
                     Strategy
                 >::apply(*it_min, *sit_min, strategy);
         }
+
+        bool is_cartesian = boost::is_same
+                <
+                    typename cs_tag<Geometry>::type,
+                    cartesian_tag
+                >::type::value;
+
+
+        bool is_polygon = boost::is_same
+                <
+                    typename tag<Geometry>::type,
+                    polygon_tag
+                >::type::value;
+
+        bool is_ring = boost::is_same
+                <
+                    typename tag<Geometry>::type,
+                    ring_tag
+                >::type::value;
+
+
+        bool is_mpolygon = boost::is_same
+                <
+                    typename tag<Geometry>::type,
+                    multi_polygon_tag
+                >::type::value;
+
+        bool is_box = boost::is_same
+                <
+                    typename tag<SegmentOrBox>::type,
+                    box_tag
+                >::type::value;
+
+        if ((is_polygon || is_mpolygon) && is_cartesian && is_box)
```

Comment:
My gut feeling tells me that `is_cartesian` should not be checked here. This should probably depend on strategy.

Reply:
Is seems that for only this case the swap of results points is needed only for cartesian geometries and I didn't want to create a more complex swap_result_points function.